test-nifikop-operator/test/atomic-1.yaml

```diff
@@ -0,0 +1,21 @@
+project: nifikop-operator
+stage: test
+mode: atomicity-violation
+front-runner: kind-control-plane
+operator-pod-label: nifikop-operator
+deployment-name: nifikop-operator
+ce-name: simplenifi
+ce-namespace: default
+ce-rtype: nificluster
+ce-diff-current: '{"spec": {"nodes": [{"readOnlyConfig": {"nifiProperties": {"overrideConfigs":
+  "nifi.ui.banner.text=NiFiKop by Sieve - Node 1\n"}, "zookeeperProperties": {}, "bootstrapProperties":
+  {}, "logbackConfig": {}, "bootstrapNotificationServicesConfig": {}}}]}}'
+ce-diff-previous: '{}'
+ce-etype-current: Updated
+ce-etype-previous: Updated
+se-name: simplenifi-config-1
+se-namespace: default
+se-rtype: secret
+se-etype: Update
+crash-location: after
```

Comment:
@laphets 
Since the crash-location is hardcoded to `before`, how to decide this one should be `after`?
More importantly, why do we need to select between `before` and `after`? For this example, if you want to crash after this side effect, why not do the crash before the next side effect?

Reply:
Good question. For general atomic bugs, I'm even not very sure whether we want crash before or after, since they look equivalent (but we may want to figure out this later). For this one, this is more like a temporary work around, since at first I didn't find the proper config generated for crash before because lack of instrument to the [status update](https://github.com/kubernetes-sigs/controller-runtime/blob/ef5c8a3ffd288667b89005aebad5ff6e0cf0c5e4/pkg/client/client.go#L305). Without noticing that when replay the bug, I choose to use the side effect we have, and crash after the it. 
The atomic mode is still work in progress, and we should get rid of the multiple crash point if we are more certain about the (se1, se2) relationship (that way, we are sure that crash after se1 or before se2 doesn't matter).

Reply:
I see. We might also support instrumenting the status update.
@laphets Could you point to me where the nifi operator uses status update (instead of the client update)?

Reply:
Yes, you can find it here: https://github.com/Orange-OpenSource/nifikop/blob/1546e0242107bf2f2c1256db50f47c79956dd1c6/pkg/k8sutil/status.go#L81

Reply:
Oh I see. Nice! Another todo for next week probably.